src/content/structured/get-started/testing-components.mdx

```diff
@@ -0,0 +1,131 @@
+
+## Installing Jest
+
+### Step one
+
+Install Jest with `npm` or `yarn`.
+
+```shell
+npm install --save-dev jest
+
+yarn add --dev jest
+```
+
+### Step two
+
+Create a test file using the extension: `.test.js` and add some tests.
+
+### Step three
+
+Add the following section to your package.json:
+
+```shell
+{
+  "scripts": {
+    "test": "jest"
+  }
+}
+```
+
+### Step four
+
+Finally, run your tests using `npm test` or `yarn test`.
+
+More information on setting up Jest can be found on [Jest’s Getting Started page](https://jestjs.io/docs/getting-started).
+
+## What is the shadow DOM?
+
+A key part of our packages is the shadow DOM. With the shadow DOM, we can encapsulate each component’s markup structure, styling and behaviour so different parts don’t clash.
+
+According to [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM), the “shadow DOM allows hidden DOM trees to be attached to elements in the regular DOM tree — this shadow DOM tree starts with a shadow root, underneath which you can attach any element, in the same way as the normal DOM.”
+
+## Testing shadow components
+
+In the DOM, elements can be found using `document.querySelector`. Shadow components can be found in a similar way using the `shadowRoot` prefix. The shadow root is the root node of the shadow DOM tree.
+
+Note: components that having been slotted in can be found in DOM. They are not a part of their parent component's shadow DOM
+
+### Example
+
+```js
+// To find all navigation items in a top nav
+document.querySelector(“ic-top-navigation).querySelectorAll(“ic-navigation-item”)
+
+// Find the text content of the first navigation item in a top nav
+document.querySelector("ic-top-navigation > ic-navigation-item:first-child").shadowRoot.querySelector("ic-tooltip > a > ic-typography").textContent
+
+// Find the search button for the search bar in the top nav (nested shadow DOM)
+document.querySelector("ic-top-navigation > ic-search-bar").shadowRoot.querySelector("ic-button#search-submit-button").shadowRoot.querySelector("#ic-button-with-tooltip-search-submit-button")
+```
+
+## Example test
+
+Below is an example of a React component that uses the `IcTopNavigation` component. The labels for the navigation items are being mapped to the `IcNavigationItem` component
+
+```js
+<---topNav.js--->
+import React from 'react';
+import { IcTopNavigation, IcNavigationItem, IcSearchBar } from "@ukic/react";
+
+export const topNav = () => {
+    const navItemLabels = ["About", "Drinks", "Recipes", "Blog"];
+
+    const navItems = navItemsLabels.map((navItemLabel, index) =>
+        {return (
+            <IcNavigationItem key={index} label={navItemLabel} href={`/${navItemLabel}`} slot="navigation">
```

Comment:
```
 const navItems = navItemsLabels.map((navItemLabel, index) => (
            <IcNavigationItem key={index} label={navItemLabel} href={`/${navItemLabel}`} slot="navigation">
```

Reply:
Why would this use regular brackets if it's an arrow function? Using the brackets clears the red error brackets I was getting but I'm not sure why this is